crates/ruff/src/message/jsonlines.rs

```diff
@@ -0,0 +1,39 @@
+use std::io::Write;
+
+use crate::message::json::message_to_json_value;
+use crate::message::{Emitter, EmitterContext, Message};
+
+#[derive(Default)]
+pub struct JsonLinesEmitter;
+
+impl Emitter for JsonLinesEmitter {
+    fn emit(
+        &mut self,
+        writer: &mut dyn Write,
+        messages: &[Message],
+        _context: &EmitterContext,
+    ) -> anyhow::Result<()> {
+        let mut w = writer;
```

Comment:
nit: don't rename here

Reply:
@konstin If I don't, I'll get

```
12 |         writer: &mut dyn Write,
   |         ------ move occurs because `writer` has type `&mut dyn std::io::Write`, which does not implement the `Copy` trait
...
17 |             serde_json::to_writer(writer, &message_to_json_value(message))?;
   |                                   ------ value moved here
18 |             writer.write_all(b"\n")?;
   |             ^^^^^^^^^^^^^^^^^^^^^^^ value borrowed here after move
```